src/python/pants/core/goals/fmt.py

```diff
@@ -101,7 +101,8 @@ def cacheable(self) -> bool:
 
 @union
 class FmtRequest(StyleRequest):
-    pass
+    # Set to `True` if this formatter makes semantic changes to the code.
+    fixer: ClassVar[bool] = False
```

Comment:
tl;dr; This comment is nothing but a rant at this point. Safe to skip.

Would it be worth being slightly more verbose here, for added clarity? (here I see the comment, but I may come across the `fixer = True` in other files, and there I will not see this comment)

To me, the implications of what `fixer = True` are is not overly clear without some detailed knowledge I have now from reading the PR description (or the preceding comment).

How about `fmt_semantically_safe: ClassVar[bool] = True` or something along those lines?
It's not something we're going to have to type a lot...

Hmm.. maybe this is a moot point to make, seeing there'll be a `--fixers` option which directly reflects onto this attribute.

I'll post this, for the record to show my train of thought here...

Reply:
I had the same thought process :) including wanting to use the word "semantic" in the classvar. But eventually went for "fixer" because of the `--fixers` option. And I went for the `--fixers` option name to keep it really short, given that we expect it to be used mostly in CLI rather than config files.

Reply:
Yeah, this is a great feature, but I'm not sure the term "fixer" will intuitively mean what we want it to mean, to a user encountering it for the first time. How about something like `--style-only` ?

Reply:
`./pants fmt --style-only ::`..I like that! I generally like when options don't start with `--no` also.

@kaos thoughts? 

(The class property could be `style_only = False`, with default to `True`)

Reply:
Yeah, `style-only` works well for me, too :) 👍🏽